### ISSUE TITLE
SQS messages without attributes cannot be received

### ISSUE THREAD

**Issue by nicosp** [aws/aws-sdk-php#2827]
### Describe the bug

Message fails with: "No Attribute MD5 found. Expected ". This issue is present between 3.285.1 and at least until 3.286.2.

I downgraded to 3.285.0 and the issue was gone.

### Expected Behavior

Receive the message without an exception.

### Current Behavior

For some reason the MessageAttributes are set and not null but they are not an array. Note that the actual message does not have any attributes or the "MessageAttributes" property.

### Reproduction Steps

Send a message without any message attributes. Try to receiveMessage.

### SDK version used

3.285.1-3.286.2

### Environment details (Version of PHP (`php -v`)? OS name and version, etc.)

PHP 8.2.12 (cli) (built: Oct 26 2023 17:23:09) (NTS) on Debian/sid

**Comment by nicosp** [comment]
An example message as received from aws cli is:
```{
  "Messages": [
    {
      "MessageId": "285d73e9-208b-455b-b082-8569c80fc613",
      "ReceiptHandle": "285d73e9-208b-455b-b082-8569c80fc613#815c4dea-98a4-4553-bdc3-b636ff4a5cab",
      "MD5OfBody": "86677f261392ee979dc608eff901ee7d",
      "Body": "<REDACTED>"
    }
  ]
}
```

**Comment by nicosp** [comment]
I think I have pinpointed the issue. We are using elasticmq for local development and when the original message does not have any MessageAttributes it sends an empty MessageAttributes array and no MD5OfMessageAttributes. This causes Sqs::calculateMessageAttributesMd5 to expect the checksum for the empty string and fails because there is no checksum in the message.

**Comment by stobrien89** [comment]
Hi @nicosp,

Dug a bit further into this and wasn't able to reproduce this using SQS.  My guess is that ElasticMq is opting to send an empty array for `MessageAttributes` in the absence of attributes.  We return `null` for empty values sent to `calculateMessageAttributesMd5`, which makes for a confusing error message, so I've updated the last condition in our SQS `validateMd5` middleware to ignore empty `MessageAttributes` values.  This should unblock ElasticMq users who are experiencing this issue.

**Comment by nicosp** [comment]
@stobrien89 Thank you for the fix. Just a heads up that elasticmq also fixed this on their side with version 1.5.2
